docs/aws/audit/kmsmonitoring/rules/secret_rotation_interval.mdx

```diff
@@ -23,6 +23,237 @@ HITRUST, AWSWAF, PCIDSS, ISO27001
+        print(f"Error enabling rotation for secret {secret_id}: {e}")
+
+# Replace 'your-secret-id' and 'your-lambda-arn' with your actual secret ID and Lambda ARN
+secret_id = 'your-secret-id'
+rotation_lambda_arn = 'your-lambda-arn'
+
+enable_secret_rotation(secret_id, rotation_lambda_arn)
+```
+
+### 3. **Schedule the Script to Run Periodically**
+To ensure that the script runs periodically, you can use a scheduling tool like cron (on Unix-based systems) or Task Scheduler (on Windows). This will help you to regularly check and enforce the rotation policy.
+
+### 4. **Monitor and Alert on Rotation Status**
+You can set up CloudWatch Alarms to monitor the rotation status of your secrets. If a secret is not rotated within the expected time frame, an alert can be triggered. Below is an example of how you can create a CloudWatch alarm using Boto3:
+
+```python
+import boto3
+
+# Initialize a session using Amazon CloudWatch
+cloudwatch = boto3.client('cloudwatch')
+
+def create_rotation_alarm(secret_id):
+    try:
+        response = cloudwatch.put_metric_alarm(
+            AlarmName=f'SecretRotationAlarm-{secret_id}',
+            MetricName='RotationAge',
+            Namespace='AWS/SecretsManager',
+            Statistic='Maximum',
+            Period=86400,  # 1 day
+            EvaluationPeriods=1,
+            Threshold=30,  # 30 days
+            ComparisonOperator='GreaterThanOrEqualToThreshold',
+            AlarmActions=[
+                'arn:aws:sns:your-region:your-account-id:your-sns-topic'  # Replace with your SNS topic ARN
+            ],
+            Dimensions=[
+                {
+                    'Name': 'SecretId',
+                    'Value': secret_id
+                },
+            ]
+        )
+        print(f"Alarm created for secret: {secret_id}")
+    except ClientError as e:
+        print(f"Error creating alarm for secret {secret_id}: {e}")
+
+# Replace 'your-secret-id' with your actual secret ID
+secret_id = 'your-secret-id'
+
+create_rotation_alarm(secret_id)
+```
+
+By following these steps, you can ensure that your secrets in AWS Secrets Manager are rotated frequently, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the AWS Key Management Service (KMS) dashboard.
+3. In the navigation pane, choose "Customer managed keys".
+4. For each KMS key listed in the dashboard, check the "Rotation status" column. If the rotation status is set to "Disabled", it means the Secrets Manager secrets are not being rotated frequently.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Secrets Manager and KMS services.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>